openssl-dynamic/src/main/java/org/apache/tomcat/jni/socket/AprSocketContext.java

```diff
@@ -797,10 +797,16 @@ void prepare() throws IOException {
         }
 
         void unblock() {
-            try (java.net.Socket sock = new java.net.Socket()) {
-                // Easiest ( maybe safest ) way to interrupt accept
-                // we could have it in non-blocking mode, etc
-                sock.connect(new InetSocketAddress("127.0.0.1", port));
+            java.net.Socket sock = new java.net.Socket();
+
+            try {
+                try {
```

Comment:
cleanup: we should only need a single `try` here. Lets move the finally to the outer `try` and remove the inner try.